### DISCUSSION TITLE
Confirm selection with enter only when completion list item is selected

### DISCUSSION THREAD

**Discussion by qazwini** [neoclide/coc.nvim#3596] [Q&A] [answered]
I currently have these configurations in `.vimrc`:
```
"inoremap <expr> <cr> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"

"Use Tab and Shift+Tab to navigate the completion list
inoremap <expr> <Tab> pumvisible() ? "\<C-n>" : "\<Tab>"
inoremap <expr> <S-TAB> pumvisible() ? "\<C-p>" : "\<S-TAB>" 
```

And the behavior that I am getting is that when the completion menu shows and I click enter, it behaves like a normal enter and creates a new line. The behavior I want is for enter to behave like a normal enter except when I have highlighted a completion list item. In that case, I want enter to hide the menu. I tried emulating this behavior using
```
inoremap <expr> <cr> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"
```
but even when I haven't highlighted a completion list item, enter just removes the completion list.

In other words, I want enter to confirm the completion if and only if I have used tab to highlight something. Otherwise, just behave like normal. Is it possible to do this?

**Comment by qazwini** [comment, marked as answer]
Found the answer. `complete_info()["selected"]` gives that info.
